File: app/src/main/java/moe/tlaster/openween/core/api/statuses/Mentions.java
package moe.tlaster.openween.core.api.statuses;

import java.util.HashMap;
import java.util.Map;

import moe.tlaster.openween.core.api.Constants;
import moe.tlaster.openween.core.model.status.MessageListModel;
import moe.tlaster.openween.core.model.types.AuthorType;
import moe.tlaster.openween.core.model.types.FeatureType;
import moe.tlaster.openween.core.model.types.SourceType;
import moe.tlaster.openween.common.helpers.HttpHelper;
import moe.tlaster.openween.common.helpers.InvalidAccessTokenException;
import moe.tlaster.openween.common.helpers.JsonCallback;

/**
 * Created by Tlaster on 2016/9/7.
 */
public class Mentions {
    public static void getMentions(long since_id, long max_id, int count, int page, AuthorType filter_by_author, SourceType filter_by_source, FeatureType filter_by_type, JsonCallback<MessageListModel> callback) throws InvalidAccessTokenException {
        Map<String, String> param = new HashMap<>();
        param.put("since_id", String.valueOf(since_id));
        param.put("max_id", String.valueOf(max_id));
        param.put("count", String.valueOf(count));
        param.put("page", String.valueOf(page));
        param.put("filter_by_author", String.valueOf(filter_by_author.getValue()));
        param.put("filter_by_source", String.valueOf(filter_by_source.getValue()));
        param.put("filter_by_type", String.valueOf(filter_by_type.getValue()));
        HttpHelper.getAsync(Constants.MENTIONS, param, callback);
    }

}

File: app/src/main/java/moe/tlaster/openween/core/api/statuses/Mentions.kt
package moe.tlaster.openween.core.api.statuses

import android.support.v4.util.ArrayMap

import moe.tlaster.openween.core.api.Constants
import moe.tlaster.openween.core.model.status.MessageListModel
import moe.tlaster.openween.core.model.types.AuthorType
import moe.tlaster.openween.core.model.types.FeatureType
import moe.tlaster.openween.core.model.types.SourceType
import moe.tlaster.openween.common.helpers.HttpHelper


/**
 * Created by Tlaster on 2016/9/7.
 */
object Mentions {
    fun getMentions(since_id: Long = 0, max_id: Long = 0, count: Int = 20, page: Int = 1, filter_by_author: AuthorType = AuthorType.All, filter_by_source: SourceType = SourceType.All, filter_by_type: FeatureType = FeatureType.All) : MessageListModel {
        val param = ArrayMap<String, String>()
        param.put("since_id", since_id.toString())
        param.put("max_id", max_id.toString())
        param.put("count", count.toString())
        param.put("page", page.toString())
        param.put("filter_by_author", filter_by_author.value.toString())
        param.put("filter_by_source", filter_by_source.value.toString())
        param.put("filter_by_type", filter_by_type.value.toString())
        return HttpHelper.getAsync(Constants.MENTIONS, param)
    }

}
